### PR TITLE
fix(nextjs): refactor how webpack config is loaded with nextjs

### LINKED ISSUE #15214: next: The @nrwl/next:build executor is incompatible with Next 13.2
### Current Behavior

When using `@nrwl/next@latest` with `next@13.2.0`, the `@nrwl/next:build` executor doesn't respect the executor options.

The `.next` folder is built inline (e.g. `apps/X/.next`) rather than in the configured `ouputPath` (e.g. `dist/apps/X/.next`).

This issue is due to an internal breaking change in version 13.2.0 of Next that removed the custom `conf` option of the `next/dist/build` function: https://github.com/vercel/next.js/pull/45455

### Expected Behavior

The `.next` build output should be located in the `outputPath` folder, e.g. `dist/apps/X/.next`.

### GitHub Repo

https://github.com/jasongerbes/nx-examples/tree/next-13.2.0

### Steps to Reproduce

1. Create a Nx workspace with `@nrwl/next`
2. Add a Next application `test-next-app`
3. Upgrade `next` to `13.2.0`
4. Run `nx build test-next-app`

### Nx Report

```shell
>  NX   Report complete - copy this into the issue template

   Node : 16.18.0
   OS   : darwin arm64
   yarn : 1.22.19
   
   nx                      : 15.7.2
   @nrwl/jest              : 15.7.2
   @nrwl/linter            : 15.7.2
   @nrwl/workspace         : 15.7.2
   @nrwl/angular           : 15.7.2
   @nrwl/cli               : 15.7.2
   @nrwl/cypress           : 15.7.2
   @nrwl/devkit            : 15.7.2
   @nrwl/eslint-plugin-nx  : 15.7.2
   @nrwl/js                : 15.7.2
   @nrwl/next              : 15.7.2
   @nrwl/react             : 15.7.2
   @nrwl/rollup            : 15.7.2
   @nrwl/tao               : 15.7.2
   @nrwl/web               : 15.7.2
   @nrwl/webpack           : 15.7.2
   @nrwl/nx-cloud          : 15.0.3
   typescript              : 4.8.4
   ---------------------------------------
   Community plugins:
   @ngrx/component-store : 15.0.0
   @ngrx/effects         : 15.0.0
   @ngrx/entity          : 15.0.0
   @ngrx/router-store    : 15.0.0
   @ngrx/store           : 15.0.0
   @ngrx/store-devtools  : 15.0.0
```

### Failure Logs

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by ndcunningham**
Currently, there is a breaking change that exists after you upgrade your `next` dependency to `next:13.2.x`

This happened because we would pass the webpack configuration to the nextJs's build function which is no longer supported

What does the mean for new / existing workspaces
- No changes are required

**TODO**

- [x] Ensure imported shared libs compiles without errors
- [x] Add e2e tests for nextjs workspaces
- [x] Rename `withNxnew` function 
- [x] Remove webpack config from `build.impl`

Fixes: #15214, https://github.com/nrwl/nx/issues/13194

### CODE REVIEW

File: packages/next/plugins/with-nx.ts [hunk 1]
```diff
@@ -34,6 +64,67 @@ function getWithNxContext(): WithNxContext {
...
+  return async (phase: PHASE, defaultConfig: NextConfig) => {
+    let dependencies: DependentBuildableProjectNode[] = [];
+
+    const graph = await createProjectGraphAsync();
+    const rootMap = createProjectRootMappings(graph.nodes);
+    const project = findProjectForPath(projectDirectory, rootMap);
+    const projectNode = graph.nodes[project];
+
+    // Assuming we know the target else we have to look it
+    const targetName = 'build-base';
```
Comment (ndcunningham) [inline packages/next/plugins/with-nx.ts:85]: will look this up based on the phase which is passed in

File: packages/next/plugins/with-nx.ts [hunk 2]
```diff
@@ -34,6 +64,67 @@ function getWithNxContext(): WithNxContext {
   };
 }
 
+// return () => Promise<NextConfig>
+export function withNxnew(
```
Comment (ndcunningham) [inline packages/next/plugins/with-nx.ts:68]: Will need to rename this to something more appropriate

**Comment by mandarini** [comment]
I used [this repo](https://github.com/dankins/nx-nextjs-13194) (taken from [the issue](https://github.com/nrwl/nx/issues/13194#issuecomment-1464513847)) as a starting point.

I built this branch locally, and made a local release, and installed it in the above repo. Also, made a number of changes to the code to reproduce initial error. 

Here is the updated repo, please read the README: https://github.com/mandarini/nx-nextjs-13194/tree/fix-1

I still got the error:

```
Server Error
Error: 

You're importing a component that needs useState. It only works in a Client Component but none of its parents are marked with "use client", so they're Server Components by default.

   ,-[1:1]
 1 | import { jsxDEV as _jsxDEV } from "react/jsx-dev-runtime";
 2 | import { useEffect, useState } from "react";
   :                     ^^^^^^^^
 3 | export const ClientComponent = ()=>{
 4 |     const [counter, setCounter] = useState(0);
 5 |     useEffect(()=>{
   `----


Maybe one of these should be marked as a client entry with "use client":
  ../client-components/src/lib/ClientComponent.tsx
  ../client-components/src/index.ts
  src/components/ReExportedComponent.tsx
  src/app/page.tsx

This error happened while generating the page. Any console logs will be displayed in the terminal window.
```

**Comment by mandarini** [comment]
Left a comment, then deleted it, because i was testing an issue that is not addressed, sorry!

**Review by mandarini — APPROVED** [review]

**Review by FrozenPandaz — APPROVED** [review]

**Comment by skaneprime** [comment]
Dunno should it be here but when building nextjs app it can't defy path correctly (ig it is next issue But not sure)

```
 info  - Checking validity of types ..Failed to compile.

../../../dist/apps/web/site/.next/types/app/layout.ts:2:24
Type error: Cannot find module '../../../../../../app/layout' or its corresponding type declarations.

  1 | // File: W:\vault\apps\web\site\app\layout.tsx
> 2 | import * as entry from '../../../../../../app/layout'
    |                        ^
  3 | import type { ResolvingMetadata } from 'next/dist/lib/metadata/types/metadata-interface'
  4 | 
  5 | type TEntry = typeof entry

 ——————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

 >  NX   Ran target build for project web-site (10s)

    ×    1/1 failed
    √    0/1 succeeded [0 read from cache]

```

```ts
// File: W:\vault\apps\web\site\app\layout.tsx
import * as entry from '../../../../../../app/layout' 
// path must be ../../../apps/subdir/project-name/app/layout 
// aka this ../../../apps/web/site/app/layout

import type { ResolvingMetadata } from 'next/dist/lib/metadata/types/metadata-interface'

type TEntry = typeof entry

// Check that the entry is a valid entry
...
```
next.config.js
```js
  typescript: {
    ignoreBuildErrors: true // will solve the issue but this is workaround. 
  },
```

**Pull request merged**